### PR TITLE
Added support for Xbox controllers over USB

### PR DESCRIPTION
**Pull request by ludokx**

### CODE REVIEW

File: backend/src/api.rs [hunk 1]
```diff
@@ -5,6 +5,8 @@ use anyhow::Result;
 use hidapi::HidApi;
 use log::debug;
 use serde::{Deserialize, Serialize};
+use std::ffi::OsStr;
+extern crate udev;
```
Comment (jfernandez) [inline backend/src/api.rs:9]: I believe `extern crate` is the legacy way to import libraries, can just use `use`. You may not even need to do this if you are calling functions with the `udev::` prefix.

File: backend/src/api.rs [hunk 2]
```diff
@@ -116,5 +117,37 @@ pub fn get_controllers() -> Result<Vec<Controller>> {
             _ => {}
         }
     }
+
+    // for Xbox over USB, hidapi-rs is not finding controllers so fall back to using udev
+    let mut enumerator = udev::Enumerator::new().unwrap();
```
Comment (jfernandez) [inline backend/src/api.rs:122]: You can replace all `.unwrap()` with `?`. unwrap will panic if there is an error and terminate the program. ? will return an error `Result` that should be caught up the chain.

**Review by jfernandez — CHANGES_REQUESTED** [review]

File: backend/src/api.rs (hunk 1 above)
Comment (ludokx) [reply backend/src/api.rs]: good catch, updated

File: backend/src/api.rs (hunk 2 above)
Comment (ludokx) [reply backend/src/api.rs]: cool thanks, updated

**Review by jfernandez — APPROVED** [review]

**Pull request merged**